blob/src/main/java/com/bazaarvoice/emodb/blob/core/DefaultBlobStore.java

```diff
@@ -299,17 +338,36 @@ private void copyTo(ByteBuffer buf, OutputStream out) throws IOException {
     }
 
     @Override
-    public void put(String tableName, String blobId, InputSupplier<? extends InputStream> in, Map<String,String> attributes) throws IOException {
+    public void put(String tableName, String blobId, InputSupplier<? extends InputStream> in, Map<String, String> attributes) throws IOException {
         checkLegalTableName(tableName);
         checkLegalBlobId(blobId);
         checkNotNull(in, "in");
         checkNotNull(attributes, "attributes");
 
         Table table = _tableDao.get(tableName);
 
+        StorageSummary summary = putBlobObject(table, blobId, in, attributes);
+
+        try {
+            _metadataProvider.writeMetadata(table, blobId, summary);
+        } catch (Throwable t) {
+            LOGGER.error("Failed to upload metadata for table: {}, blobId: {}, attempt to delete blob. Exception: {}", tableName, blobId, t.getMessage());
+
+            try {
+                _storageProvider.deleteObject(table, blobId);
+            } catch (Exception e1) {
+                LOGGER.error("Failed to delete blob for table: {}, blobId: {}. Inconsistency between blob and metadata storages. Exception: {}", tableName, blobId, e1.getMessage());
+                _metadataInconsistencyMeter.mark();
```

Comment:
In this instance, the marker is marked when we have the blob data but write metadata failed, deletion of data failed, but comment for the marker said opposite:
     * This meter was created to provide visibility for inconsistency
     * when metadata storage contains data but blob storage doesn't.

Reply:
right, will fix